cmd/main.go

```diff
@@ -0,0 +1,211 @@
+}
+
+func fileExists(filePath string) (bool, error) {
+	if _, err := os.Stat(filePath); err == nil {
+		return true, nil
+	} else if errors.Is(err, os.ErrNotExist) {
+		return false, nil
+	} else {
+		return false, err
+	}
+}
+
+func Install() error {
+	assetDownloadPath := downloadLatestReleaseAsset()
+	defer os.Remove(assetDownloadPath)
+	switch runtime.GOOS {
+	case "darwin":
+		// Use native unzip tool as it handles the creation of required symbolic links
+		cmd := exec.Command("unzip", assetDownloadPath, "-d", "/Applications")
+
+		if out, err := cmd.CombinedOutput(); err != nil {
+			log.Println(string(out))
+			return err
+		}
+
+		return nil
+	case "windows":
+		installDir, err := getInstallDir()
+		if err != nil {
+			return err
+		}
+		if err := os.Mkdir(installDir, os.ModePerm); err != nil {
+			return err
+		}
+
+		// tar utility is available from Windows build 17063
+		// TODO: consider using other command or a custom implementation
+		// https://learn.microsoft.com/en-us/virtualization/community/team-blog/2017/20171219-tar-and-curl-come-to-windows
+		cmd := exec.Command("tar", "-xf", assetDownloadPath, "-C", installDir)
+
+		if out, err := cmd.CombinedOutput(); err != nil {
+			log.Println(string(out))
+			return err
+		}
+
+		return nil
+	default:
+		return errorPlatformNotSupported
+	}
+}
+
+func downloadLatestReleaseAsset() string {
+	release, error := getLatestRelease()
+
+	if error != nil {
+		panic(error)
+	}
+
+	resp, _ := http.Get(*release.asset.BrowserDownloadURL)
+	defer resp.Body.Close()
+
+	out, _ := os.Create(*release.asset.Name)
+	defer out.Close()
+
+	log.Printf("Downloading latest Flowser release %s", release.version)
+
+	_, err := io.Copy(out, resp.Body)
+	if err != nil {
+		panic("Install failed")
+	}
+
+	return out.Name()
+}
+
+type FlowserRelease struct {
+	version string
+	asset   github.ReleaseAsset
+}
+
+func getLatestRelease() (FlowserRelease, error) {
+	client := github.NewClient(nil)
+	release, _, err := client.Repositories.GetLatestRelease(context.Background(), "onflowser", "flowser")
+	if err != nil {
+		log.Fatal(err)
+	}
+	latestVersion := strings.Replace(*release.TagName, "v", "", 1)
+	targetAssetName, err := getAssetName(latestVersion)
+	if err != nil {
+		return FlowserRelease{}, err
+	}
+	for _, asset := range release.Assets {
+		if *asset.Name == targetAssetName {
+			return FlowserRelease{
+				version: latestVersion,
+				asset:   asset,
+			}, nil
+		}
+	}
+	return FlowserRelease{}, errors.New("no asset found")
+}
+
+func getAssetName(version string) (string, error) {
+	isArm := strings.HasPrefix(runtime.GOARCH, "arm")
+	switch runtime.GOOS {
+	case "darwin":
+		if isArm {
+			return fmt.Sprintf("Flowser-%s-arm64-mac.zip", version), nil
+		} else {
+			return fmt.Sprintf("Flowser-%s-mac.zip", version), nil
+		}
+	case "windows":
+		return fmt.Sprintf("Flowser-%s-win.zip", version), nil
+	default:
+		return "", errorPlatformNotSupported
+	}
+}
+
+func getInstallDir() (string, error) {
```

Comment:
the function should take in the path as first arg and platform as the second, then it should return path if it's provided or get the default as state bellow.